azure-pipelines.yml

```diff
@@ -147,6 +147,10 @@ jobs:
       name: NetCoreInternal-Int-Pool
       queue: buildpool.windows.10.amd64.vs2017
 
+    variables:
+      ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+        _OfficialBuildIdArgs: /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
```

Comment:
I don't know how to define this variable in a single spot. I don't really speak yaml. So if anyone has a suggestion on how define this variable once in the yaml file and reuse it across both jobs, I'm all ears.

Reply:
Can't you define it in the global variables section?

Reply:
I did that, and it failed to start the build. I'm sure I did something wrong, but I assume it had something to do with the `    ${{ if ` at the beginning. Do those always need to go last in the list?

Reply:
That is what I thought, I don't think you can globally defined variables with a condition, I don't think it is allowed because the global section is expanded before evaluating the ${{ expressions at compile time.

Reply:
We could technically use a variable template, but I don't think it is worth to just avoid duplicating 1 variable.